### PR TITLE
WIP: NEUBURG-27: Use cityConfig API

### PR DESCRIPTION
**Pull request by NKlug**
This pull request belongs to an issue on our [bugtracker](https://integreat.atlassian.net/). 
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **WEBAPP**.

### CODE REVIEW

**Comment by NKlug** [comment]
Some tests are still failing and I can't figure out why exactly. Any ideas?

File: src/modules/endpoint/endpoints/categories.js [hunk 1]
```diff
@@ -12,7 +12,9 @@ function getUrlWithoutCity (url) {
 }
 
 export default new EndpointBuilder('categories')
-  .withStateToUrlMapper(state => `https://cms.integreat-app.de/${getCurrentCityConfig().cmsName}/de/wp-json/extensions/v0/modified_content/pages?since=1970-01-01T00:00:00Z`)
+  .withStateToUrlMapper(state => {
+    return `https://cms.integreat-app.de/${getCurrentCityConfig(state.cityConfigs._data).cmsName}/de/wp-json/extensions/v0/modified_content/pages?since=1970-01-01T00:00:00Z`
```
Comment (NKlug) [inline src/modules/endpoint/endpoints/categories.js]: Is there any other way to retrieve the cityConfigs from the state than `state.cityConfigs._data`?

**Comment by NKlug** [comment]
And I think we have to enable some kind of cors in the backend. I couldn't get it working without this but also don't have much experience with cors, does anyone have any comments on this?

File: src/modules/endpoint/endpoints/categories.js (hunk 1 above)
Comment (michael-markl) [reply src/modules/endpoint/endpoints/categories.js]: As far as I can see, with the current setup of the withFetcher: no.

**Comment by michael-markl** [comment]
For the cors you can install a addon https://addons.mozilla.org/en-US/firefox/addon/cors-everywhere or https://chrome.google.com/webstore/detail/allow-control-allow-origi/nlfbmbojpeacfghkpbjhddihlkkiljbi
We already set it up for api.wohnen.integreat-app.de - the addon is only necessary for your local environment.

File: src/modules/app/containers/RouterFragment.js [hunk 2]
```diff
@@ -8,11 +8,16 @@ import LivingManageOfferPage from '../../../routes/living-manage-offer/container
 import LivingLayout from '../../layout/containers/LivingLayout'
 import DisclaimerPage from '../../../routes/disclaimer/containers/DisclaimerPage'
 import getCurrentCityConfig from '../../city-detection/getCurrentCityConfig'
+import withFetcher from 'modules/endpoint/hocs/withFetcher'
 import { Helmet } from 'react-helmet'
+import CityConfig from '../../city-detection/CityConfig'
+import compose from 'lodash/fp/compose'
+import connect from 'react-redux/es/connect/connect'
```
Comment (michael-markl) [inline src/modules/app/containers/RouterFragment.js]: This is the problem for the failing tests:
Use `import { connect } from 'rect-redux'` instead
Edit: This doesn't fix them, but change the error message :D Should help you fix it though

**Review by michael-markl — APPROVED** [review]

**Comment by NKlug** [comment]
> For the cors you can install a addon https://addons.mozilla.org/en-US/firefox/addon/cors-everywhere or https://chrome.google.com/webstore/detail/allow-control-allow-origi/nlfbmbojpeacfghkpbjhddihlkkiljbi
> We already set it up for api.wohnen.integreat-app.de - the addon is only necessary for your local environment.

Does it work for you with this browser extension? I already have it installed and it still didn't work until I enabled cors in the backend.

**Comment by michael-markl** [comment]
Yes, it did work, you also have to activate it once it's installed, did you do that as well? 🤔

**Comment by NKlug** [comment]
I'm pretty sure I activated it :D 
Maybe I configured something else incorrectly.

**Comment by michael-markl** [comment]
Ok, yeah idk^^ for me the add-on is green when I activate it in the toolbar (of chrome) - maybe I didn't try Firefox yet

**Comment by NKlug** [comment]
Just reinstalled it, now it works :)

**Comment by maxammann** [comment]
Another way would be to proxy the server locally. More work to do initially but works probably better :)

**Pull request merged**